src/utils/utils.ts

```diff
@@ -447,328 +337,90 @@ export const orderToJSON = (order: Order): OrderJSON => {
-  // Zero sometimes returned as 0x from contracts
-  if (arg === "0x") {
-    arg = 0;
-  }
-  // fix "new BigNumber() number type has more than 15 significant digits"
-  arg = arg.toString();
-  return new BigNumber(arg);
-}
-
-/**
- * Estimate Gas usage for a transaction
- * @param web3 Web3 instance
- * @param from address sending transaction
- * @param to destination contract address
- * @param data data to send to contract
- * @param value value in ETH to send with data
+ * Estimate gas usage for a transaction.
+ * @param provider The Provider
+ * @param from Address sending transaction
+ * @param to Destination contract address
+ * @param data Data to send to contract
+ * @param value Value in ETH to send with data
  */
 export async function estimateGas(
-  web3: Web3,
+  provider: ethers.providers.Provider,
   { from, to, data, value = 0 }: TxData
-): Promise<number> {
-  const amount = await web3.eth.estimateGas({
+) {
+  return await provider.estimateGas({
     from,
     to,
     value: value.toString(),
     data,
   });
-
-  return amount;
 }
 
 /**
- * Get mean gas price for sending a txn, in wei
- * @param web3 Web3 instance
+ * Get mean gas price for sending a txn, in wei.
+ * @param provider The provider
  */
-export async function getCurrentGasPrice(web3: Web3): Promise<BigNumber> {
-  const gasPrice = await web3.eth.getGasPrice();
-  return new BigNumber(gasPrice);
-}
-
-/**
- * Get current transfer fees for an asset
- * @param web3 Web3 instance
- * @param asset The asset to check for transfer fees
- */
-export async function getTransferFeeSettings(
-  web3: Web3,
-  {
-    asset,
-    accountAddress,
-  }: {
-    asset: Asset;
-    accountAddress?: string;
-  }
-) {
-  let transferFee: BigNumber | undefined;
-  let transferFeeTokenAddress: string | undefined;
-
-  if (asset.tokenAddress.toLowerCase() == ENJIN_ADDRESS.toLowerCase()) {
-    // Enjin asset
-    const feeContract = new web3.eth.Contract(
-      ERC1155,
-      asset.tokenAddress
-    ) as unknown as ERC1155Abi;
-
-    const params = await feeContract.methods
-      .transferSettings(asset.tokenId as string)
-      .call({ from: accountAddress });
-    if (params) {
-      transferFee = makeBigNumber(params[3]);
-      if (params[2] === "0") {
-        transferFeeTokenAddress = ENJIN_COIN_ADDRESS;
-      }
-    }
-  }
-  return { transferFee, transferFeeTokenAddress };
+export async function getCurrentGasPrice(provider: ethers.providers.Provider) {
+  return provider.getGasPrice();
 }
 
 /**
  * Estimates the price of an order
  * @param order The order to estimate price on
  * @param secondsToBacktrack The number of seconds to subtract on current time,
  *  to fix race conditions
- * @param shouldRoundUp Whether to round up fractional wei
  */
-export function estimateCurrentPrice(
-  order: Order,
-  secondsToBacktrack = 30,
-  shouldRoundUp = true
-) {
+export function estimateCurrentPrice(order: Order, secondsToBacktrack = 30) {
```

Comment:
Are we sure all the numbers changed to `BigNumber.from` here are whole numbers?

Reply:
removing this now-unused fn